xls/examples/bitonic_sort.x

```diff
@@ -0,0 +1,220 @@
+////////////////////////////////////////////////////////////////////////////////
+// Bitonic sort
+// A template function for sort an array of u32.
+// The array size must be a power of two.
+// The template takes the log2 of the size of the array as parameter.
+//
+// For example, assuming "array" is of type u32[8].
+// Then, bitonic_sort<u32:3>(array) will return the sorted array.
+////////////////////////////////////////////////////////////////////////////////
+
+fn bitonic_sort<LOG_N: u32, N: u32 = { u32:1 << LOG_N }>(array: u32[N]) -> u32[N] {
```

Comment:
curious if you can use `std::clog2`?

Reply:
more specifically:
```
fn bitonic_sort<N: u32, LOG_N: u32 = { std::clog2(N) }>(array: u32[N]) -> u32[N]
```
that could allow the function to infer N directly from the argument.

Reply:
Yes, it works. The only issue is that users can pass non-power-of-two sizes.
I guess we need a static_assert on parameters.

Reply:
Yeah, I tried "fail!", but could not get it to work as I wanted. I will try again.